openEO web editor first look (#139)
eu-cdse/documentation: Applications/WebEditor.qmd

---
title: "openEO Web Editor"
---

The openEO Web Editor is a web-based graphical user interface (GUI) that allows users to interact with the openEO API and perform various tasks related to Earth observation data processing. It provides a user-friendly interface for users who are not familiar with a programming language to carry out several Earth Observation data processing tasks, such as querying available data, defining processing workflows, executing processes, and visualising the results. It allows users to build complex processing chains by connecting different processing steps as building blocks and provides options to specify parameters and input data for each step. 

> The openEO Web Editor can be accessed via [https://openeo.dataspace.copernicus.eu/](https://openeo.dataspace.copernicus.eu/){target="_blank"}. Even without logging in, users have the ability to retrieve information on available collections, processes, User Defined Functions(UDF) Runtimes, and the options for exporting files. Additionally, users can create process building blocks; however, to execute them, logging in is necessary.

Reviewer: ```suggestion
The openEO Web Editor can be accessed via [https://openeo.dataspace.copernicus.eu/](https://openeo.dataspace.copernicus.eu/){target="_blank"}. Even without logging in, users have the ability to retrieve information on available collections, processes, User Defined Functions (UDF) Runtimes, and the options for exporting files. Additionally, users can create openEO process graphs, however, log in is necessary to execute them.
```

5. **Guest**

    The *Guest* naming will be replaced with your username when logged in. The dropdown will provide with an option to Log in.

6. **Features**

    The basic functionalities that can be handy when creating the processes in openEO Web Editor is available in this row. These functionalities includes creating a new script, importing processes from external sources, exporting in another programming language, validating processes on the server side, editing process metadata, adding parameters, etc. 

7. **Process Editor**
   
   This is the editor for the processes. We recommend to work in "**Visual Model**" mode, where you can create processing chains simply by adding collections and processes and connecting them with each other. The "**Code**" mode allows to see the generated JSON process, which is usually only needed if you want to run the process using another client library such as Python or R.

   The area on it's right will later be used for previewing collections or inspecting the results of batch jobs, web services or other computations. It will also be used to display log messages, if available.

8.  **Log in**

>     As previously mentioned, it is necessary to log in to interact with the server. A new window will appear when attempting to log in, as demonstrated below. There are multiple methods available to authenticate with the server. Choose the option for which you possess valid credentials. Typically, the recommended choice is the Copernicus Data Space Ecosystem option. However, you also have the option to log in using EGI Check-in, where you can select your organisation account. For further information regarding various authentication methods or to seek assistance, you can always click on the "help" option at the top or contact us.

Reviewer: ```suggestion
    As previously mentioned, it is necessary to log in to interact with the server. A new window will appear when attempting to log in, as demonstrated below. While other options are sometimes available, the recommended authentication choice is the "Copernicus Data Space Ecosystem". For further information regarding various authentication methods or to seek assistance, you can always click on the "help" option at the top or contact us.
```

>     <centre><img src="_images/login.PNG"  width="80%" height="60%"></centre>

Reviewer: Note: we'll have to redo this screenshot once EGI is removed as option

Author: Sure.

Author: updated
